tests/test_point/test_point_arithmetic.py

```diff
@@ -0,0 +1,106 @@
+import contextlib
+
+import pytest
+from cryptography.hazmat.backends.openssl import backend
+from cryptography.hazmat.primitives.asymmetric import ec
+
+from umbral.bignum import BigNum
+from umbral.config import default_curve
+from umbral.point import Point
+
+curve = default_curve() or ec.SECP256K1()
+
+
+@pytest.fixture()
+def mock_openssl(mocker, random_ec_point1, random_ec_bignum):
+    @contextlib.contextmanager
+    def mocked_openssl_backend():
+        def mocked_ec_point_equality(group, ec_point, other_point, bn_ctx):
+            assert 'BN_CTX' in str(bn_ctx)
+            assert 'EC_GROUP' in str(group)
+            assert 'EC_POINT' in str(ec_point)
+            assert 'EC_POINT' in str(other_point)
+
+            assert random_ec_point1.group == group
+            assert random_ec_point1.ec_point == ec_point
+            assert random_ec_point1.ec_point == other_point
+
+            return 1
+
+        def mocked_ec_point_addition(group, sum, ec_point, other_point, bn_ctx):
+            assert 'BN_CTX' in str(bn_ctx)
+            assert 'EC_POINT' in str(other_point)
+            assert 'EC_POINT' in str(sum)
+
+            assert random_ec_point1.group == group
+            assert random_ec_point1.ec_point == ec_point
+
+            return 1
+
+        def mocked_ec_point_multiplication(group, product, null, ec_point, bignum, bn_ctx):
+            assert 'BN_CTX' in str(bn_ctx)
+            assert 'EC_GROUP' in str(group)
+            assert 'EC_POINT' in str(ec_point)
+            assert 'EC_POINT' in str(product)
+            assert 'NULL' in str(null)
+
+            assert random_ec_point1.group == group
+            assert random_ec_point1.ec_point == ec_point
+
+            assert random_ec_bignum.bignum == bignum
+            assert random_ec_bignum.curve_nid == random_ec_point1.curve_nid
+
+            return 1
+
+        def mocked_ec_point_inversion(group, inverse, bn_ctx):
+            assert 'EC_POINT' in str(inverse)
+            assert 'BN_CTX' in str(bn_ctx)
+
+            assert random_ec_point1.group == group
+
+            return 1
+
+        mock_load = {'EC_POINT_mul': mocked_ec_point_multiplication,
+                     'EC_POINT_cmp': mocked_ec_point_equality,
+                     'EC_POINT_add': mocked_ec_point_addition,
+                     'EC_POINT_invert': mocked_ec_point_inversion}
+
+        with contextlib.ExitStack() as stack:
+            for method, patch in mock_load.items():
+                stack.enter_context(mocker.mock_module.patch.object(backend._lib, method, patch))
+            yield
+
+    return mocked_openssl_backend
+
+
+def test_mocked_openssl_point_arithmetic(mock_openssl, random_ec_point1, random_ec_point2, random_ec_bignum):
+    with mock_openssl():
+        _ = random_ec_point1 == random_ec_point1   # __eq__
```

Comment:
Instead of merely saving and releasing these, why not make assertions about them?

Reply:
Possible, but we would have to use the actual openssl methods to generate bool expressions in each case.